### ISSUE TITLE
Sometimes SIP fails to make the PVD of the mask

### ISSUE THREAD

**Issue by paoloserra** [kmhess/SoFiA-image-pipeline#35]
I'm getting this error.

```
	-Source 1: SoFiA J035043.00-361253.9.
	Start making spatial images.
	No SPECSYS or VELREF in header to define reference frame, checking CTYPE3.
	Found CTYPE3 spectral axis type VRAD in header.
	No SPECSYS, VELREF, or reference frame in CTYPE3, assuming data in TOPOCENT reference frame.
	Found J2000.0 equinox in header.
	The first HI contour defined at SNR = [2.5, 3.5] has level = 3.965e+01 (mom0 data units).
	No user image given. Proceeding with the download of any requested archive images.
WARNING: AstropyDeprecationWarning: The truth value of a Quantity is ambiguous. In the future this will raise a ValueError. [astropy.units.quantity]
WARNING: InputWarning: Coordinate string is being interpreted as an ICRS coordinate. [astroquery.utils.commons]
	Survey image retrieved from DSS2 Blue.
	Making DSS2 Blue overlaid with HI contours.
	Making HI grey scale map.
	Making pixel SNR map.
	Making velocity map.
	WARNING: Assuming channels are uniform width in velocity.
	WARNING: Assuming channels are uniform width in velocity.
	Making pv slice.
	Reading in mask cubelet.
Traceback (most recent call last):
  File "/home/pserra/.local/lib/python3.7/site-packages/pvextractor/utils/wcs_utils.py", line 19, in get_spatial_scale
    np.testing.assert_almost_equal(abs(scale[0,0]), abs(scale[0,1]))
  File "/home/pserra/.local/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 599, in assert_almost_equal
    raise AssertionError(_build_err_msg())
AssertionError: 
Arrays are not almost equal to 7 decimals
 ACTUAL: 0.0055558091777239815
 DESIRED: 0.00555530281073768

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pserra/Astro/software/SoFiA-image-pipeline/image_pipeline.py", line 175, in <module>
    surveys=list(surveys), snr_range=args.snr_range, user_image=args.user_image, user_range=args.user_range)
  File "/home/pserra/Astro/software/SoFiA-image-pipeline/src/make_images.py", line 608, in main
    make_pv(source, src_basename, cube_params, opt_view=opt_view, suffix=suffix)
  File "/home/pserra/Astro/software/SoFiA-image-pipeline/src/make_images.py", line 405, in make_pv
    mask_pv = create_pv(source, src_basename + '_{}_mask.fits'.format(str(source['id'])), opt_view=opt_view[0])
  File "/home/pserra/Astro/software/SoFiA-image-pipeline/modules/functions.py", line 285, in create_pv
    mask_pv = extract_pv_slice(mask[0].data, slice, wcs=WCS(mask[0].header, fix=True, translate_units='shd'))
  File "/home/pserra/.local/lib/python3.7/site-packages/pvextractor/pvextractor.py", line 78, in extract_pv_slice
    scale = get_spatial_scale(wcs)
  File "/home/pserra/.local/lib/python3.7/site-packages/pvextractor/utils/wcs_utils.py", line 21, in get_spatial_scale
    raise ValueError("Non-square pixels.  Please resample data.")
ValueError: Non-square pixels.  Please resample data.
```

**Comment by paoloserra** [comment]
> It's good to not be too broad, in catching exceptions, if possible, right?

I agree, but I'm clearly no programmer :)

**Comment by kmhess** [comment]
If you change AssertionError to ValueError, does it work?

**Comment by paoloserra** [comment]
Just to clarify, I think this deserves to be merged into master as it allows users with this problem to keep going.

Maybe we can open a separate issue about how to solve the problem and find a way to plot the mask contours.

**Comment by kmhess** [comment]
Branch has been merged.  I'm happy to keep this as the open issue since it has your error at the top.  I did a little digging into  the Assertion Error, but I think that might be a feature that astropy needs to deal with, tbh.

**Comment by paoloserra** [comment]
OK, I will rename the issue though.

**Comment by kmhess** [comment]
For sample data in Galactic coordinates, the mask PVD also fails.

**Comment by kmhess** [comment]
I've noticed this issue seems to have resolved itself somehow with the #57 noregrid branch, at least for some sources!

**Comment by kmhess** [comment]
It looks like pvextractor may allow a work around for this issue now with a new PR:
https://github.com/radio-astro-tools/pvextractor/pull/113
Some background:
https://github.com/radio-astro-tools/pvextractor/issues/107
